metrics/engine/ingester.go

```diff
@@ -92,6 +97,74 @@ func (oi *outputIngester) flushMetrics() {
 				oi.metricsEngine.markObserved(sm.Metric)
 				sm.Metric.Sink.Add(sample)
 			}
+
+			oi.cardinality.Add(sample.TimeSeries)
 		}
 	}
+
+	if hit, limit := oi.cardinality.LimitHit(); hit {
+		// TODO: suggest using the Metadata API as an alternative, once it's
+		// available (e.g. move high-cardinality tags as Metadata)
+		// https://github.com/grafana/k6/issues/2766
+
+		oi.logger.Warnf("The test has generated %d time series, which is higher than the suggested limit (%d), "+
+			"and could cause high memory usage. "+
+			"Consider not using high-cardinality values like IDs as tags "+
+			"or if you need them in the URL then use the URL-grouping feature. "+
+			"See https://k6.io/docs/using-k6/tags-and-groups for details.", oi.cardinality.Count(), limit)
```

Comment:
```suggestion
		oi.logger.Warnf(
			"The test has generated metrics with %d unique time series, "+
			"which is higher than the suggested limit of %d "+
			"and could cause high memory usage. "+
			"Consider not using high-cardinality values like unique IDs as metric tags "+
			"or, if you need them in the URL, use the name metric tag or URL grouping. "+
			"See https://k6.io/docs/using-k6/tags-and-groups for details.", oi.cardinality.Count(), timeSeriesFirstLimit)
```

Slightly edited the message and specified the limit to be `timeSeriesFirstLimit` - if we hit this message multiple times (e.g. at 100k, 200k, 400k), it doesn't make sense to me to have multiple warnings where we give a different value for the "suggested limit".